File: java-code/src/main/java/refactoring/extract_function/ExtractFunction_After.java
package refactoring.extract_function;

import lombok.extern.slf4j.Slf4j;
import refactoring.model.Invoice;
import refactoring.model.Order;

@Slf4j
public class ExtractFunction_After {

  public void printOwing(Invoice invoice) {

    printBanner();
    var outstanding = calculateOutstanding(invoice);
    printDetails(invoice, outstanding);
  }

  private static int calculateOutstanding(Invoice invoice) {
    var outstanding = 0;

    for (Order order : invoice.getOrders()) {
      outstanding += order.getAmount();
    }
    return outstanding;
  }

  private static void printDetails(Invoice invoice, int outstanding) {
    // print details
    log.info("name: " + invoice.getCustomer());
    log.info("amount: " + outstanding);
    log.info("due: " + invoice.getDueDate());
  }

  void printBanner() {
    log.info("*************************");
    log.info("***** Customer Owes *****");
    log.info("*************************");
  }
}

File: go-code/basics/extract-function/extract-function-after.go
package extract_function

import (
	"fmt"
	"gitlab.com/clean-code55/refactor/improving-design-of-existing-code/go-code/basics/models"
)

func printOwingAfter(invoice models.Invoice) {
	printBanner()
	outstanding := calculateOutstanding(invoice)
	printDetails(invoice, outstanding)
}

func calculateOutstanding(invoice models.Invoice) int {
	outstanding := 0

	for _, order := range invoice.Orders {
		outstanding += order.Amount
	}
	return outstanding
}

func printDetails(invoice models.Invoice, outstanding int) {
	fmt.Println("name: " + invoice.Customer)
	fmt.Printf("amount:  %d\n", outstanding)
	fmt.Printf("due: %v\n", invoice.DueDate)
}

func printBanner() {
	fmt.Println("************************")
	fmt.Println("***Customer Owes***")
	fmt.Println("************************")
}
